client/src/main/java/org/apache/rocketmq/client/impl/producer/TopicPublishInfo.java

```diff
@@ -29,6 +29,15 @@
     private List<MessageQueue> messageQueueList = new ArrayList<MessageQueue>();
     private volatile ThreadLocalIndex sendWhichQueue = new ThreadLocalIndex();
     private TopicRouteData topicRouteData;
+    private long notFoundFlag = -1;
+
+    public void setNotFoundFlag() {
+        if (!checkNotFoundFlag()) {
+            notFoundFlag = System.currentTimeMillis();
+        }
+    }
+
+    public boolean checkNotFoundFlag() { return System.currentTimeMillis() < notFoundFlag + 2000; }
```

Comment:
It is easier to understand the definition of constants instead of magic values 2000.

Reply:
See the next commit